rust-lang/cargo
# doc: clarify config-relative paths for `--config <path>`

### What does this PR try to resolve?

Resolves #10991. 

Instead of changing the behaviour (it might break things), we document the current one as @sunshowers [suggesting](https://github.com/rust-lang/cargo/issues/10991#issuecomment-1215931242).

### How should we test and review this PR?

Check changes in 

- `src/doc/src/reference/config.md` and 
- `src/doc/man/includes/section-options-common.md`.

```shell
cargo help build
# Proofread the manpage change.
mdbook serve
# open http://[::1]:3000/reference/config.html#config-relative-paths
# and proofread the config-relative paths section
```
